src/libraries/Pool.sol

```diff
@@ -383,10 +394,12 @@ library Pool {
 
             // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (cache.protocolFee > 0) {
-                // calculate the amount of the fee that should go to the protocol
-                uint256 delta = step.feeAmount * cache.protocolFee / ProtocolFeeLibrary.BIPS_DENOMINATOR;
-                // subtract it from the regular fee and add it to the protocol fee
                 unchecked {
+                    // calculate the amount of the fee that should go to the protocol
+                    // fee amount has already been subtracted from the amountIn
```

Comment:
i think the (step.amountIn + step.feeAmount) addition is still a little misleading & want the comment to be ultra clear

maybe something like 
// step.amountIn does not include the swap fee. we add it back to get the total amountIn to calculate the amount of fees owed to the protocol


not really sure?? open to suggestions I just find this part of the line extremely confusing `(step.amountIn + step.feeAmount)`